src/bt_peer_connection.cpp

```diff
@@ -366,24 +411,17 @@ namespace libtorrent {
 #ifndef TORRENT_DISABLE_LOGGING
 		peer_log(peer_log_alert::outgoing_message, "HAVE_ALL");
 #endif
-		static const char msg[] = {0,0,0,1, msg_have_all};
-		send_buffer(msg, sizeof(msg));
-
-		stats_counters().inc_stats_counter(counters::num_outgoing_have_all);
+		send_simple_message(msg_have_all, counters::num_outgoing_have_all);
 	}
 
 	void bt_peer_connection::write_have_none()
 	{
 		INVARIANT_CHECK;
-		TORRENT_ASSERT(m_sent_handshake);
 		m_sent_bitfield = true;
 #ifndef TORRENT_DISABLE_LOGGING
 		peer_log(peer_log_alert::outgoing_message, "HAVE_NONE");
 #endif
-		static const char msg[] = {0,0,0,1, msg_have_none};
-		send_buffer(msg, sizeof(msg));
-
-		stats_counters().inc_stats_counter(counters::num_outgoing_have_none);
+		send_simple_message(msg_have_none, counters::num_outgoing_have_none);
 	}
 
 	void bt_peer_connection::write_reject_request(peer_request const& r)
```

Comment:
write_reject_request() could use the ``send_request_message()`` function too, right?